meson.build

```diff
@@ -1,27 +1,42 @@
-project(
-    'io.elementary.terminal',
-    'vala', 'c',
-    version: '6.1.1'
-)
+project('io.elementary.terminal', 'vala', 'c', version: '6.1.1', meson_version: '>=0.58')
+
+valac = meson.get_compiler('vala')
+cc = meson.get_compiler('c')
 
-if meson.get_compiler('vala').version().version_compare('<0.40')
+if valac.version().version_compare('<0.40')
     error('Missing required dependency: valac >= 0.40')
 endif
 
 gnome = import('gnome')
 i18n = import('i18n')
 
-add_global_arguments([
-        '-DGETTEXT_PACKAGE="@0@"'.format(meson.project_name()),
-        '-DHANDY_USE_UNSTABLE_API'
-    ],
-    language:'c'
-)
+glib_version = '>=2.40'
+gtk_version = '>-3.24'
+granite_version = '>=6.1'
+handy_version = '>=0.83'
+vte_version = '>=0.59'
```

Comment:
Is there a good reason to have these all separated out like this? it seems like it kind of just makes it more complicated and I'm not sure we ever want to re-use these values besides `glib_version`

Reply:
i would say it make more easy to find minimal versions of dependencies, but i can inline the not re-used ones if preferred.